src/main.rs

```diff
@@ -19,6 +21,75 @@ use hidreport::hid::{
 };
 use hidreport::*;
 
+static mut OUTFILE: OnceLock<
+    std::sync::Mutex<std::cell::RefCell<std::io::LineWriter<std::fs::File>>>,
+> = OnceLock::new();
+
+enum Outfile {
+    Stdout,
+    File(&'static mut std::sync::Mutex<std::cell::RefCell<std::io::LineWriter<std::fs::File>>>),
+}
+
+impl Outfile {
+    fn new() -> Self {
+        unsafe {
+            match OUTFILE.get_mut() {
+                None => Outfile::Stdout,
+                Some(o) => Outfile::File(o),
+            }
+        }
+    }
+
+    fn init(cli: &Cli) -> Result<()> {
+        if cli.output_file == "-" {
+            // Bit lame but easier to just set the env for owo_colors to figure out the rest
+            match cli.color {
+                ColorChoice::Never => std::env::set_var("NO_COLOR", "1"),
+                ColorChoice::Auto => {}
+                ColorChoice::Always => std::env::set_var("FORCE_COLOR", "1"),
+            }
+
+            return Ok(());
+        }
+
+        std::env::set_var("NO_COLOR", "1");
```

Comment:
unrelated note because it's an existing bug: this does not handle `--color=always` together with an output file. meh, I guess but still...

Reply:
filed #28 so we don't lose it